test/Fail.hs

```diff
@@ -0,0 +1,47 @@
+{-# LANGUAGE FlexibleContexts, RankNTypes, ScopedTypeVariables, TypeApplications #-}
+module Fail
```

Comment:
`Fail` is technically a synonym of `Throw`, but we have a separate carrier for it which wasn’t being tested, and it’s typically used via the `MonadFail` interface instead of via `throwError`. This module rectifies the above lapses.